### DISCUSSION TITLE
Is it possible to define environment variables as results of shell commands ?

### DISCUSSION THREAD

**Discussion by obourdon** [goreleaser/goreleaser#1634] [Q&A] [answered]
First, thanks a lot for this wonderful tool. I have unsuccessfully tried to define something like follows in by .goreleaser.yml file:
```
  env:
   - CGO_ENABLED=0
  - GIT_BRANCH=$(git rev-parse --abbrev-ref HEAD || echo 'n/a')
  - GIT_REVISION=$(git describe --always --tags --dirty || echo 'n/a')
  - GO_BUILD_DATE=$(date -u +%FT%T)
  ldflags: -s -w -X main.GitCommit={{.Env.GIT_REVISION}}
```
however it seems like this does not work (error during go  build with bad parameter in link phase) and was not able to find any clue in the issues/discussions/code of this repository

**Comment by caarlos0** [comment, marked as answer]
inside the yaml its not possible, no.

you can do it before running goreleaser though:

```sh
GIT_BRANCH=$(git rev-parse --abbrev-ref HEAD || echo 'n/a') GIT_REVISION=$(git describe --always --tags --dirty || echo 'n/a') GO_BUILD_DATE=$(date -u +%FT%T) goreleaser release --rm-dist
```

also note that the revision and date are already available as template variables: https://goreleaser.com/customization/templates/
